src/main/java/com/exactpro/th2/act/ActHandler.java

```diff
@@ -578,6 +642,43 @@ private void sendMessage(Message message, EventID parentEventId) throws IOExcept
         }
     }
 
+    private void multiSendMessage(List<Message> messages, EventID parentEventId) throws IOException {
+        try {
+            LOGGER.debug("Sending the message started");
+
+            if (messages.isEmpty()) {
+                throw new IllegalArgumentException("Empty message list is not allowed");
+            }
+
+            final var firstMessageId = messages.get(0).getMetadata().getId();
+            final var bookName = firstMessageId.getBookName();
+            final var sessionGroup = firstMessageId.getConnectionId().getSessionGroup();
+
+            // currently all messages should be placed into one batch, and therefore they should have
+            // the same 'bookName' and 'sessionGroup' fields
+            // later we can implement batching based on book name and session group
+            if(!messages.stream().allMatch(msg -> msg.getMetadata().getId().getBookName().equals(bookName)
+                    && msg.getMetadata().getId().getConnectionId().getSessionGroup().equals(sessionGroup))) {
+                throw new IllegalArgumentException("All messages in 'multiSendMessage' should have the same 'bookName' (" + bookName + ") and 'sessionGroup' (" + sessionGroup + ") fields");
+            }
+
+            var batch = new GroupBatch(bookName, sessionGroup, messages.stream()
+                    .map(msg -> toGroup(MessageUtilsKt.toTransportBuilder(msg).setEventId(EventUtilsKt.toTransport(parentEventId)).build()))
+                    .collect(Collectors.toList())
+            );
+
+            messageRouter.send(batch, SEND_QUEUE_ATTRIBUTE);
+            //TODO remove after solving issue TH2-217
+            //TODO process response
+/*            EventBatch eventBatch = EventBatch.newBuilder()
+                    .addEvents(createSendMessageEvent(parsedMessage, parentEventId))
+                    .build();
+            eventBatchMessageRouter.send(eventBatch, "publish", "event");*/
```

Comment:
I believe the ticket TH2-217 is already solved and we can either remove this code completely or uncomment it